public/components/datasources/components/manage/data_connection.tsx

```diff
@@ -294,6 +344,55 @@ export const DataConnection = (props: any) => {
     );
   };
 
+  useEffect(() => {
+    // Loads databases cache when cache is empty
+    if (datasourceDetails.name) {
+      const cache = CatalogCacheManager.getDataSourceCache().dataSources.find(
+        (datasource) => datasource.name === datasourceDetails.name
+      );
+      if (cache?.status !== 'Updated') {
+        startLoadingDatabases(datasourceDetails.name);
+        setIsRefreshing(true);
+      } else {
+        const cachedList: CachedDatabase[] =
+          CatalogCacheManager.getDataSourceCache().dataSources.find(
+            (cachedDataSource) => cachedDataSource.name === datasourceDetails.name
+          )?.databases || [];
+        if (cachedList) {
+          setCachedDatabases(cachedList);
+        }
+        setIsRefreshing(false);
+        setIsFirstTimeLoading(false);
+      }
+    }
+  }, [datasourceDetails.name]);
+
+  useEffect(() => {
+    // Handles loading tables and acceleration cache when cache is empty
+    const loadingStatuses = [DirectQueryLoadingStatus.RUNNING, DirectQueryLoadingStatus.WAITING];
```

Comment:
move to a const file or move to the top of the file? we don't need this every useEffect runs